src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java

```diff
@@ -0,0 +1,636 @@
+import org.isf.utils.jobjects.ModalJFrame;
+import org.isf.utils.time.TimeTools;
+import org.isf.ward.manager.WardBrowserManager;
+import org.isf.ward.model.Ward;
+import org.springframework.data.domain.Page;
+
+public class InventoryWardBrowser extends ModalJFrame implements InventoryListener {
+
+    private static final long serialVersionUID = 1L;
+
+    private GoodDateChooser jCalendarTo;
+    private GoodDateChooser jCalendarFrom;
+    private LocalDateTime dateFrom = TimeTools.getDateToday0();
+    private LocalDateTime dateTo = TimeTools.getDateToday24();
+    private JLabel jLabelTo;
+    private JLabel jLabelFrom;
+    private JPanel panelHeader;
+    private JPanel panelFooter;
+    private JPanel panelContent;
+    private JButton closeButton;
+    private JButton newButton;
+    private JButton editButton;
+    private JButton deleteButton;
+    private JButton viewButton;
+    private JScrollPane scrollPaneInventory;
+    private JTable jTableInventory;
+    private String[] pColums = {
+            MessageBundle.getMessage("angal.inventory.referenceshow.col").toUpperCase(),
+            MessageBundle.getMessage("angal.common.ward.col").toUpperCase(),
+            MessageBundle.getMessage("angal.common.date.col").toUpperCase(),
+            MessageBundle.getMessage("angal.inventory.status.col").toUpperCase(),
+            MessageBundle.getMessage("angal.common.user.col").toUpperCase()
+    };
+    private int[] pColumwidth = { 150, 150, 100, 100, 150 };
+    private JComboBox<String> statusComboBox;
+    private JLabel statusLabel;
+    JButton next;
+    JButton previous;
+    JComboBox<Integer> pagesCombo = new JComboBox<>();
+    JLabel under = new JLabel("/ 0 " + MessageBundle.getMessage("angal.inventory.page.text"));
+    private static int PAGE_SIZE = 50;
+    private int startIndex;
+    private int totalRows;
+    private MedicalInventoryManager medicalInventoryManager = Context.getApplicationContext().getBean(MedicalInventoryManager.class);
+    private WardBrowserManager wardBrowserManager = Context.getApplicationContext().getBean(WardBrowserManager.class);
+    private List<MedicalInventory> inventoryList;
+
+    public InventoryWardBrowser() {
+        initComponents();
+    }
+
+    private void initComponents() {
+
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        setMinimumSize(new Dimension(850, 550));
+        setLocationRelativeTo(null); // center
+        setTitle(MessageBundle.getMessage("angal.inventory.managementward.title"));
+
+        panelHeader = getPanelHeader();
+        getContentPane().add(panelHeader, BorderLayout.NORTH);
+
+        panelContent = getPanelContent();
+        getContentPane().add(panelContent, BorderLayout.CENTER);
+
+        panelFooter = getPanelFooter();
+        getContentPane().add(panelFooter, BorderLayout.SOUTH);
+
+        ajustWidth();
+
+        addWindowListener(new WindowAdapter() {
+            public void windowClosing(WindowEvent e) {
```

Comment:
Add missing @Override annotation

Reply:
not applicable here, the method is just called, not defined.

Reply:
sorry, you meant on windowClosing method, done. Thanks!